### PR TITLE
Fix position in reports for CSS and JS in HTML

### LINKED ISSUE #1956: [Bug] CSS hints report wrong position for CSS inline in HTML [3.5]
# 🐛 Bug report

## Description

Save and open the following HTML document in VS Code with the webhint extension:
```html
<!DOCTYPE html>
<html>
    <head>
        <style>            
            .example {
                appearance: none;
                -webkit-appearance: none;
                -moz-appearance: none;
            }
        </style>
    </head>
</html>
```

Expected: Error reported at `appearance: none`.

Actual: Error reported at `<html>`

Affects both `hint-compat-api` and `hint-css-prefix-order`. Should be able to be fixed using additional event data from PR #1953.

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

Fix #1956 

Includes a number of improvements across packages already split into dedicated commits.

### CODE REVIEW

File: packages/hint/src/lib/types/html.ts [hunk 1]
```diff
@@ -79,39 +79,89 @@ export class HTMLElement {
...
+    public getContentLocation(offset: ProblemLocation): ProblemLocation | null {
+        const location = this._getOriginalLocation();
+
+        if (!location) {
+            return null;
+        }
+
+        // Get the end of the start tag from `parse5`, converting to be zero-based.
+        const startTag = location.startTag;
+        const column = startTag.endCol - 1;
```
Comment (sarvaje) [inline packages/hint/src/lib/types/html.ts]: Is this ok? We are not converting the column in the method `getLocation`

Comment (antross) [reply packages/hint/src/lib/types/html.ts]: Yes, in `getLocation` per the comment there we want to skip past the `<` and point to the tag name. So it's kind of like we're adding one then converting the column.

Here we want to point to the start of the element content after `>` which is what `parse5` already gives us so we need to convert the position.

Comment (sarvaje) [reply packages/hint/src/lib/types/html.ts]: ok!

**Comment by antross** [comment]
@sarvaje @molant @utsavized This is ready for review if you have some time. 

I'd like to get it in so I can rebase #1966 on it.

File: packages/hint-sri/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "ava": {
     "failFast": false,
     "files": [
-      "dist/tests/**/*.js"
+      "dist/tests/**/tests-https.js"
```
Comment (sarvaje) [inline packages/hint-sri/package.json]: Did you forget to restore this change?

Comment (antross) [reply packages/hint-sri/package.json]: I did, but it's fixed now (caught it right after I mentioned everyone to look at the PR 🤦‍♂️)

**Comment by antross** [comment]
@sarvaje Any other feedback? Or am I good to merge this?

**Comment by sarvaje** [comment]
@antross LGTM

**Pull request merged**